gravitee-apim-portal-webui-next/src/app/catalog/catalog.component.spec.ts

```diff
@@ -199,11 +208,82 @@ describe('CatalogComponent', () => {
     });
   });
 
-  function expectApiList(apisResponse: ApisResponse = fakeApisResponse(), page: number = 1, size: number = 18, q: string = '') {
-    httpTestingController.expectOne(`${TESTING_BASE_URL}/apis/_search?page=${page}&category=&size=${size}&q=${q}`).flush(apisResponse);
+  describe('Category filters', () => {
+    const CATEGORY_1 = fakeCategory({ id: 'category-1', name: 'Category 1' });
+    const CATEGORY_2 = fakeCategory({ id: 'category-2', name: 'Category 2' });
+    describe('With no categories', () => {
+      beforeEach(async () => {
+        await init();
+        expectApiList();
+        expectCategoriesList(fakeCategoriesResponse({ data: [] }));
+      });
+      it('should not show filters if no categories', async () => {
+        const categoryTabs = await harnessLoader.getHarnessOrNull(MatTabGroupHarness);
+        expect(categoryTabs).toBeNull();
+      });
+    });
+
+    describe('With no filter defined in params', () => {
+      beforeEach(async () => {
+        await init();
+        expectApiList();
+        expectCategoriesList(fakeCategoriesResponse({ data: [CATEGORY_1, CATEGORY_2] }));
+      });
+      it('should categories + All as filters', async () => {
+        const tabs = await harnessLoader.getAllHarnesses(MatTabHarness);
+        expect(tabs).toHaveLength(3);
+        expect(await tabs[0].getLabel()).toEqual('All');
+        expect(await tabs[1].getLabel()).toEqual(CATEGORY_1.name);
+        expect(await tabs[2].getLabel()).toEqual(CATEGORY_2.name);
+      });
+      it('should navigate to category', async () => {
+        const category1Tab = await harnessLoader.getHarness(MatTabHarness.with({ label: CATEGORY_1.name }));
+        await category1Tab.select();
+
+        expect(routerNavigateSpy).toHaveBeenCalledWith([''], {
+          queryParams: { filter: CATEGORY_1.id, query: undefined },
+          relativeTo: expect.anything(),
+        });
+      });
+    });
+
+    describe('With specified filter in params', () => {
+      beforeEach(async () => {
+        await init(CATEGORY_2.id);
+        expectApiList(undefined, undefined, undefined, undefined, CATEGORY_2.id);
+        expectCategoriesList(fakeCategoriesResponse({ data: [CATEGORY_1, CATEGORY_2] }));
```

Comment:
Looks like we can extracted dublicated code:
`  const setup = async (categoryId: string = '') => {
    await init(categoryId);
    expectApiList();
    expectCategoriesList(fakeCategoriesResponse({ data: categoryId ? [CATEGORY_1, CATEGORY_2] : [] }));
  };`

and here it could be used like this:
`await setup(CATEGORY_2.id);`

And on previous before statement, it can be used like this:
`await setup();`

Reply:
Done